logging.handlers.WatchedFileHandler: cannot use keyword arguments?

```python
from logging.handlers import WatchedFileHandler

wfh = WatchedFileHandler('foo.log', delay=True)
```

```
$ python3 --version
Python 3.6.8

$ mypy --version
mypy 0.740

$ mypy test1.py
test1.py:3: error: No overload variant of "WatchedFileHandler" matches argument types "str", "bool"
test1.py:3: note: Possible overload variants:
test1.py:3: note:     def WatchedFileHandler(self, filename: Union[str, _PathLike[str]]) -> WatchedFileHandler
test1.py:3: note:     def WatchedFileHandler(self, filename: Union[str, _PathLike[str]], mode: str) -> WatchedFileHandler
test1.py:3: note:     <2 more similar overloads not shown, out of 4 total overloads>
```

In `typeshed`, I see [4 overloads][1] suggesting positional use with 1..4 arguments, but surely passing `delay` as a keyword argument is a legitimate use?

[1]: https://github.com/python/typeshed/blob/master/stdlib/2and3/logging/handlers.pyi#L31-L41

Python 2.7 docs [document this constructor as taking 1..4 positional parameters][2] but 3.5 and above [simply specify all parameters beyond filename to have default values][3].

[2]: https://docs.python.org/2/library/logging.handlers.html#watchedfilehandler
[3]: https://docs.python.org/3.5/library/logging.handlers.html#watchedfilehandler

Would the following be okay:

```python
class WatchedFileHandler(FileHandler):
    def __init__(self, filename: _Path, mode: str = ...,
                 encoding: Optional[str] = ...,
                 delay: bool = ...) -> None: ...
```

or would it have to be split by Python version?

```python
class WatchedFileHandler(FileHandler):
    if sys.version_info >= (3,):
        def __init__(self, filename: _Path, mode: str = ...,
                     encoding: Optional[str] = ...,
                     delay: bool = ...) -> None: ...
    else:
        @overload
        def __init__(self, filename: _Path) -> None: ...
        @overload
        def __init__(self, filename: _Path, mode: str) -> None: ...
        @overload
        def __init__(self, filename: _Path, mode: str,
                     encoding: Optional[str]) -> None: ...
        @overload
        def __init__(self, filename: _Path, mode: str, encoding: Optional[str],
                     delay: bool) -> None: ...

--- a/stdlib/2and3/logging/handlers.pyi
+++ b/stdlib/2and3/logging/handlers.pyi
@@ -29,16 +29,8 @@ SYSLOG_UDP_PORT: int
 SYSLOG_TCP_PORT: int
 
 class WatchedFileHandler(FileHandler):
-    @overload
-    def __init__(self, filename: _Path) -> None: ...
-    @overload
-    def __init__(self, filename: _Path, mode: str) -> None: ...
-    @overload
-    def __init__(self, filename: _Path, mode: str,
-                 encoding: Optional[str]) -> None: ...
-    @overload
-    def __init__(self, filename: _Path, mode: str, encoding: Optional[str],
-                 delay: bool) -> None: ...
+    def __init__(self, filename: _Path, mode: str = ..., encoding: Optional[str] = ...,
+                 delay: bool = ...) -> None: ...
 
 
 if sys.version_info >= (3,):

